File: backend-springboot/src/main/java/br/com/portalproposta/backend/controller/SupervisorController.java
package br.com.portalproposta.backend.controller;

import br.com.portalproposta.backend.exception.AppBadRequestException;
import br.com.portalproposta.backend.model.PedidoAnalise;
import br.com.portalproposta.backend.model.VerifyPasswordRequest;
import br.com.portalproposta.backend.model.VerifyPasswordResponse;
import br.com.portalproposta.backend.service.ConfiguracaoService;
import br.com.portalproposta.backend.service.SupervisorService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.net.URI;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/supervisor")
public class SupervisorController {

    private final SupervisorService supervisorService;
    private final ConfiguracaoService configuracaoService;

    @Autowired
    public SupervisorController(SupervisorService supervisorService, ConfiguracaoService configuracaoService) {
        this.supervisorService = supervisorService;
        this.configuracaoService = configuracaoService;
    }

    @PostMapping("/verify-password")
    public ResponseEntity<VerifyPasswordResponse> verifyPassword(@RequestBody VerifyPasswordRequest req) {
        try {
            configuracaoService.verifySupervisorPassword(req.getPassword());
            return ResponseEntity.ok(new VerifyPasswordResponse(true));
        } catch (AppBadRequestException e) {
            return ResponseEntity.ok(new VerifyPasswordResponse(false));
        }
    }

    @GetMapping("/requests")
    public ResponseEntity<List<PedidoAnalise>> listRequests(@RequestParam(required = false) String status) {
        List<PedidoAnalise> requests;
        if ("Pendente".equals(status)) {
            requests = supervisorService.listPendentes();
        } else {
            requests = supervisorService.listAll();
        }
        return ResponseEntity.ok(requests);
    }

    @PostMapping("/requests")
    public ResponseEntity<PedidoAnalise> createRequest(@RequestBody PedidoAnalise request) {
        PedidoAnalise created = supervisorService.createRequest(request);
        URI uri = ServletUriComponentsBuilder.fromCurrentRequest()
                .path("/{id}")
                .buildAndExpand(created.getId())
                .toUri();
        return ResponseEntity.created(uri).body(created);
    }

    @PostMapping("/requests/{id}/approve")
    public ResponseEntity<Map<String, String>> approveRequest(@PathVariable String id) {
        supervisorService.approveRequest(id);
        Map<String, String> response = new HashMap<>();
        response.put("message", "Solicitação aprovada e executada com sucesso");
        return ResponseEntity.ok(response);
    }

    @PostMapping("/requests/{id}/reject")
    public ResponseEntity<Map<String, String>> rejectRequest(@PathVariable String id) {
        supervisorService.rejectRequest(id);
        Map<String, String> response = new HashMap<>();
        response.put("message", "Solicitação rejeitada com sucesso");
        return ResponseEntity.ok(response);
    }
}

File: backend/Controllers/SupervisorController.cs
using Microsoft.AspNetCore.Mvc;
using PortalProposta.Models;
using PortalProposta.Services;
using System.Text.Json.Serialization;

namespace PortalProposta.Controllers;

[ApiController]
[Route("api/supervisor")]
public class SupervisorController(SupervisorService supervisorService, ConfiguracaoService configService) : ControllerBase
{
    public record VerifyPasswordRequest([property: JsonPropertyName("password")] string Password);
    public record VerifyPasswordResponse([property: JsonPropertyName("valid")] bool Valid);

    [HttpPost("verify-password")]
    public async Task<ActionResult<VerifyPasswordResponse>> VerifyPassword(VerifyPasswordRequest req)
    {
        try
        {
            await configService.VerifySupervisorPasswordAsync(req.Password);
            return Ok(new VerifyPasswordResponse(true));
        }
        catch (AppBadRequestException)
        {
            return Ok(new VerifyPasswordResponse(false));
        }
    }

    [HttpGet("requests")]
    public async Task<ActionResult<List<PedidoAnalise>>> ListRequests([FromQuery] string? status)
    {
        var requests = status == "Pendente"
            ? await supervisorService.ListPendentesAsync()
            : await supervisorService.ListAllAsync();
        return Ok(requests);
    }

    [HttpPost("requests")]
    public async Task<ActionResult<PedidoAnalise>> CreateRequest(PedidoAnalise request) =>
        CreatedAtAction(nameof(ListRequests), await supervisorService.CreateRequestAsync(request));

    [HttpPost("requests/{id}/approve")]
    public async Task<IActionResult> ApproveRequest(string id)
    {
        await supervisorService.ApproveRequestAsync(id);
        return Ok(new { message = "Solicitação aprovada e executada com sucesso" });
    }

    [HttpPost("requests/{id}/reject")]
    public async Task<IActionResult> RejectRequest(string id)
    {
        await supervisorService.RejectRequestAsync(id);
        return Ok(new { message = "Solicitação rejeitada com sucesso" });
    }
}
